driver/src/main/java/org/neo4j/driver/internal/async/pool/NettyChannelTracker.java

```diff
@@ -65,26 +65,27 @@ public void channelAcquired( Channel channel )
     @Override
     public void channelCreated( Channel channel )
     {
-        log.debug( "Channel %s created", channel );
-        incrementInUse( channel );
-        metricsListener.afterCreated( serverAddress( channel ) );
+        throw new IllegalStateException( "A fatal error happened in this driver as this method should never be called. " +
```

Comment:
Maybe just `throw new UnsupportedOperationException()`? Current message reminds me of `ThisShouldNotHappenError` :)